docs/content/latest/contribute/core-database/coding-style.md

```diff
@@ -0,0 +1,525 @@
+---
+title: YugabyteDB coding style
+headerTitle: YugabyteDB coding style
+linkTitle: Coding style
+description: YugabyteDB coding style
+image: /images/section_icons/index/quick_start.png
+headcontent: YugabyteDB coding style
+type: page
+menu:
+  latest:
+    identifier: coding-style
+    parent: core-database
+    weight: 2914
+isTocNested: true
+showAsideToc: true
+---
+
+YugabyteDB is primarily written in C++, with some parts of the build system and test suite written in Python, Java, and Bash (to be replaced with Python).
+We also use [Protocol Buffers](https://developers.google.com/protocol-buffers) to define some data and network message formats.
+
+## Language-agnostic coding style
+
+### Variable names
+
+Avoid rarely used abbreviations. Think about whether all other potential readers of the code know about the abbreviation you are about to use.
+
+### Comments
+
+When writing full sentences, start them with a capital letter and end them with a period. This rule does not apply if the comment is just a single phrase on the same line with a code statement.
+
+## C coding style
+For the modified PostgreSQL C codebase residing inside the YugabyteDB codebase we adhere to [PostgreSQL Coding Conventions](https://www.postgresql.org/docs/13/source-format.html).
+Note that PostgreSQL code uses tabs for indentation, but we use spaces for indentation everywhere else in YugabyteDB code.
+
+## C++ coding style
+
+### Line length
+
+Use 100-character line length limit instead of 80-character.
+
+### Formatting function declarations and definitions
+
+Use one of the following formatting styles for function declarations and definitions.
+
+#### All arguments on one line
+
+```c++
```

Comment:
Tagging the language as `c++` won't get source-highlighted. This is visible in the build preview: https://deploy-preview-8837--infallible-bardeen-164bc9.netlify.app/latest/contribute/core-database/coding-style/#all-arguments-on-one-line

````suggestion
```cpp
````

Reply:
Fixed.